static/js/shared/place_selector.tsx

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import _ from "lodash";
+import React, { useEffect, useState } from "react";
+import { Card, Container, CustomInput } from "reactstrap";
+
+import { SearchBar } from "../tools/timeline/search";
+import {
+  ENCLOSED_PLACE_TYPE_NAMES,
+  getNamedTypedPlace,
+  getParentPlacesPromise,
+} from "../utils/place_utils";
+import { NamedTypedPlace } from "./types";
+
+const EMPTY_NAMED_TYPED_PLACE = { dcid: "", name: "", types: null };
+
+interface PlaceSelectorProps {
+  selectedPlace: NamedTypedPlace;
+  enclosedPlaceType: string;
+  onPlaceSelected: (place: NamedTypedPlace) => void;
+  onEnclosedPlaceTypeSelected: (type: string) => void;
+  getEnclosedPlaceTypes: (
+    place: NamedTypedPlace,
+    parentPlaces: NamedTypedPlace[]
+  ) => string[];
+  customSearchPlaceholder?: string;
+  children?: React.ReactNode;
+}
+
+export function PlaceSelector(props: PlaceSelectorProps): JSX.Element {
+  const [childPlaceTypes, setChildPlaceTypes] = useState([]);
+  useEffect(() => {
+    if (_.isNull(props.selectedPlace.types)) {
+      return;
+    }
+    loadChildPlaceTypes();
+  }, [props.selectedPlace]);
+
+  return (
+    <Card className="place-options-card">
+      <Container className="place-options" fluid={true}>
+        <div className="place-options-section" id="place-search-section">
+          <div className="place-options-label">Plot places in</div>
+          <div id="search">
```

Comment:
while we're refactoring - would be nice to have a prefix used for ids/classes in a module. that would reduce collisions across modules. wdyt?

Reply:
I don't fully understand what you mean by this. Do you mean have a const string for each module and append that string to the beginning of each id/class in the module? And by module, do you mean each component?

Reply:
initial thought was to keep it consistent - most classes here are place-options-* but this div has id=search (which could be place-options-search). since we're refactoring, we can add some consistency. i like the idea of having a const string to ensure a consistent prefix.

yes. by module, i meant this component.

Reply:
done